scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,172 @@
+     * @param index the stack offsets of the variable's lifetime(s)
+     * @param variableName the name of the local variable
+     */
+    public LocalLocation(Label[] start, Label[] end, int[] index, String variableName) {
+        this.start = start;
+        this.end = end;
         this.index = index;
-        this.scopeStart = scopeStart;
-        this.scopeLength = scopeLength;
-        this.varName = null;
-        this.varIndex = -1;
+        this.variableName = variableName;
     }
 
-    public LocalLocation(String varName, int varIndex) {
-        this.index = -1;
-        this.scopeStart = -1;
-        this.scopeLength = -1;
-        this.varName = varName;
-        this.varIndex = varIndex;
+    /**
+     * Construct a new LocalLocation representing a single scope/lifetime.
+     * Only being used by Writers, not Readers for now. Should possibly deprecate this in the future.
+     * Changes values reflectively.
+     *
+     * @param index the offset of the variable in the stack frame
+     * @param scopeStart the bytecode offset of the start of the variable's lifetime
+     * @param scopeLength the bytecode length of the variable's lifetime
+     */
+    public LocalLocation(int index, int scopeStart, int scopeLength) {
+        Label startLabel = new Label();
+        Label endLabel = new Label();
+
+        try {
+            Field flagsField = Label.class.getDeclaredField("flags");
+            Field bytecodeOffsetField = Label.class.getDeclaredField("bytecodeOffset");
+            Field FLAG_RESOLVED_FIELD = Label.class.getDeclaredField("FLAG_RESOLVED");
+
+            flagsField.setAccessible(true);
+            bytecodeOffsetField.setAccessible(true);
+            FLAG_RESOLVED_FIELD.setAccessible(true);
+            int FLAG_RESOLVED = (Integer) FLAG_RESOLVED_FIELD.get(null);
+
+            short flags = (Short) flagsField.get(startLabel);
+            flags |= FLAG_RESOLVED;
+            flagsField.set(startLabel, flags);
+            bytecodeOffsetField.set(startLabel, scopeStart);
+
+            flags = (Short) flagsField.get(endLabel);
+            flags |= FLAG_RESOLVED;
+            flagsField.set(endLabel, flags);
+            bytecodeOffsetField.set(endLabel, scopeStart + scopeLength);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        this.start = new Label[] {startLabel};
+        this.end = new Label[] {endLabel};
+        this.index = new int[] {index};
+        this.variableName = null;
     }
 
+    /**
+     * Returns the bytecode offset to the start of the first scope/lifetime.
+     *
+     * @return the bytecode offset to the start of the first scope/lifetime
+     */
+    public int getScopeStart() {
+        try {
+            return start[0].getOffset();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(start));
+            return -1;
```

Comment:
The possibility of returning -1 is not documented.

Also, when is the exception caught, and is that a bad situation?  Should this throw an exception rather than continuing?

Reply:
I've documented -1.  As to the exception - this code was inherited from KushG.  While I'm not completely familiar with all the possibilities, it does seem like a fatal problem to me too.

Reply:
I suggest making a separate branch in which you make this throw an exception.
 * If no tests fail in that branch, you can merge that branch into this one, and remove the documentation about -1.
 * If tests fail in that branch, you can investigate the problem.

Reply:
FYI, I did this in my branch `noasmx-unresolvedlabel`, and the exception does get thrown.